Source/WTF/wtf/StdLibExtras.h

```diff
@@ -1540,6 +1540,15 @@ ALWAYS_INLINE void lazyInitialize(const std::unique_ptr<T>& ptr, std::unique_ptr
     const_cast<std::unique_ptr<T>&>(ptr) = std::move(obj);
 }
 
+ALWAYS_INLINE double stringToDouble(std::span<const char> buffer, size_t& length)
```

Comment:
I'd call the out parameter parsedLength.

Re `errno`, I would suggest either:
- Use `errno` as the error return mechanism, and rename this function to `strtodSpan` (to call out that this is the buffer-safe version, but you still have `strtod` `errno` semantics)
- Use an optional return value as the error return mechanism, and keep the name `stringToDouble` (to make it more WebKit-y),